utils/hare-fetch-fids

```diff
@@ -0,0 +1,164 @@
+    return f'{container:#x}:{key:#x}'
+
+
+def node_names(data: List) -> List[Node]:
+    # Get entries from file
+    # ​"key": "m0conf/nodes/<node_fid>",
+    # "value": "{​​​​​\"name\": \"<node name>\", \"state\": \"<HA state>\"}​​​​​"
+    regex = re.compile(
+        f'm0conf/nodes/[a-zA-Z0-9:]+$')
+    nodes = []
+    for key in data:
+        match_result = re.match(regex, key['key'])
+        if not match_result:
+            continue
+        if len(key['key'].split('/')) == 3:
+            nodes.append(json.loads(key['value'])['name'])
+    return nodes
+
+
+def get_keys_for_processes(data: List) -> List[str]:
+    # Get 'm0conf/nodes/<node_id>/processes/<process_fidk>/services/<svc_type>' entries
+    # from the file.  See 'Consul KV Schema' in [4/KV](rfc/4/README.md).
+    regex = re.compile(
+        f'^m0conf\\/.*\\/processes\\/.*\\/services\\/(.+)$')
+    keys = []
+    for key in data:
+        match_result = re.match(regex, key['key'])
+        if not match_result:
+            continue
+        keys.append(match_result.group(0))
+    return keys
+
+
+def add_service_to_node(nodes: List[Node], node_name: str, svc_name: str, fid: str):
+    for node in nodes:
+        if node.name == node_name:
+            node.add_service(svc_name, fid)
+
+
+def get_service_for_node(nodes: List[Node], node_name: str, service: str) -> str:
+    for node in nodes:
+        if node.name == node_name:
+            return node.get_service(service)
+
+
+def process_all_keys(keys: List, nodes: List[Node]):
+    names = {
+        'confd': 'confd',
+        'ios': 'ioservice',
+        'm0_client_s3': 's3server'
+    }
+    for key in keys:
+        # key looks like
+        # 'm0conf/nodes/<name>/processes/<process_fidk>/services/<svc_type>'
+        parts = key.split('/')
+        svc_type = parts[-1]
+        if svc_type in names:
+            node_name = parts[2]
+            process_fidk = parts[-3]
+            add_service_to_node(nodes, node_name, names[svc_type],
+                                to_fid(0x7200000000000001, int(process_fidk)))
+
+
+def read_file():
+    path = '/var/lib/hare/consul-kv.json'
+    with open(path) as consul_kv_file:
+        data = json.load(consul_kv_file)
+    return data
+
+
+def get_nodes_for_cluster() -> List[Node]:
+    file_data = read_file()
+    nodes = []
+    for n in node_names(file_data):
+        nodes.append(Node(n))
+
+    keys = get_keys_for_processes(file_data)
+    process_all_keys(keys, nodes)
+    return nodes
+
+
+def parse_opts(argv):
+    p = argparse.ArgumentParser(description='Fetches the fids for services configured on a node.',
+                                usage='%(prog)s [OPTION]')
+
+    p.add_argument('--service', '-s',
+                   help='service name.'
+                   'List of services- "confd", "ioservice", "s3server".',
+                   required=True,
+                   type=str,
+                   action='store')
+    p.add_argument('--node', '-n',
```

Comment:
Now that we are using consul-kv.json that has information for all the nodes in the cluster, why not keep `node` as an optional argument and by default the interface returns the fids of processes for the current node (from where the interfaces is invoked).
Let's add a `--all` option to return fids for all the nodes in the cluster.
When a `node` is provided, we fetch fids for the given node.

Reply:
Yes, done.  Here is the summary.

```
NODE   SERVICE    RESULT
YES        YES      (FID OF GIVEN SERVICE OF GIVEN NODE)
NO        YES       (FID OF GIVEN SERVICE OF CURRENT NODE)
ALL                 (RETURN ALL)
NO        NO        (RETURN FIDS FOR ALL SERVICES OF CURRENT NODE)
YES        NO       (RETURN FIDS FOR ALL SERVICES OF GIVEN NODE)
```

Reply:
Awesome!